pkg/bloomgateway/bloomgateway.go

```diff
@@ -161,9 +161,18 @@ func (g *Gateway) stopping(_ error) error {
 	return services.StopManagerAndAwaitStopped(context.Background(), g.serviceMngr)
 }
 
-func (g *Gateway) PrefetchBloomBlocks(_ context.Context, _ *logproto.PrefetchBloomBlocksRequest) (*logproto.PrefetchBloomBlocksResponse, error) {
-	// TODO: Implement prefetching of bloom blocks
-	return &logproto.PrefetchBloomBlocksResponse{}, nil
+func (g *Gateway) PrefetchBloomBlocks(ctx context.Context, req *logproto.PrefetchBloomBlocksRequest) (*logproto.PrefetchBloomBlocksResponse, error) {
+	refs, err := decodeBlockKeys(req.Blocks)
+	if err != nil {
+		return nil, err
+	}
+	_, err = g.bloomStore.FetchBlocks(
```

Comment:
I think we should check the blocks and close the onces that may be opened. This should not happen but maybe we end up using this API for sth else in the future
```go
for _, bq := range bqs {
		if bq != nil {
			// This is the expected case: the blocks is not yet downloaded and the block querier is nil
			continue
		}

		// Close any block querier that were already downloaded
		if err := bq.Close(); err != nil {
			level.Error(g.logger).Log("msg", "failed to close block querier", "err", err)
		}
	}
```